test/JDBC/expected/TestDecimal-vu-verify.out

```diff
@@ -265,3 +265,11 @@ select in3 * in4 from testdecimal_vu_prepare_tab2;
 
 ~~ERROR (Message: Arithmetic overflow error for data type numeric.)~~
 
+
+#BABEL-3066
+SELECT CAST(123456.25 AS real) AS value_as_real, CAST(CAST(123456.25 AS real) AS decimal(30,2)) AS value_casted_to_decimal_30_2;
```

Comment:
Please add more test cases where this casting can happen. When a column is one of type, doing insert, insert from column of one type to another type and so on